### DISCUSSION TITLE
GetArguments vs GetArgument on 1.0rc0001 and rc0002

### DISCUSSION THREAD

**Discussion by DTTerastar** [cake-build/cake#3026] [Q&A]
I tested my stuff on 1.0rc0001 and rc0002 and the lack of Arguments.GetArgument is a big blocker: Error: System.AggregateException: One or more errors occurred. (Method not found: 'System.String Cake.Core.ICakeArguments.GetArgument(System.String)'.)
---> System.MissingMethodException: Method not found: 'System.String Cake.Core.ICakeArguments.GetArgument(System.String)'.
at Cake.Utils.Prompt.PromptAliases.ArgumentT
at Submission#0.ArgumentT

The fact 0.38 and 0.37 don't have a GetArguments also means I can't use start using that or else my addon will fail on that version.

Would a PR with a GetArgument being a call to GetArguments().SingleOrDefault() be met w/ happiness or disdain? It can be Obsolete from the start.

**Comment by patriksvensson** [comment]
@DTTerastar You could change the addin to use the `Argument`-extension method in `Cake.Common` (https://github.com/cake-build/cake/blob/develop/src/Cake.Common/ArgumentAliases.cs#L74), which should be the same in pre1.0 and 1.0.

Another option could be to invoke the method via reflection, and if it fails, call the other one.

**Comment by devlead** [comment]
What's your user case / the problem you're trying to solve?

Could you use [Argument<T>(ICakeContext, string, T)](https://cakebuild.net/api/Cake.Common/ArgumentAliases/5C55EC67) which is existed for several versions on a higher level in Common?

**Comment by pascalberger** [comment]
> Would a PR with a GetArgument being a call to GetArguments().SingleOrDefault() be met w/ happiness or disdain? It can be Obsolete from the start.

@DTTerastar We would accept a PR which adds a `GetArgument` method back as an extension to `ICakeArguments`. Would you be interested in providing a PR?

**Reply by DTTerastar** [reply]
Sure I can dot that.

**Reply by pascalberger** [reply]
I created https://github.com/cake-build/cake/issues/3029 for tracking this

**Reply by pascalberger** [reply]
A `GetArguments` extension has been added with https://github.com/cake-build/cake/pull/3049 and will be shipped as part of the next 1.0 release.

**Comment by augustoproiete** [comment]
An extension method called `GetArgument` shipped with [Cake v1.0](https://cakebuild.net/blog/2021/02/cake-v1.0.0-released):

```csharp
var helloArg = Context.Arguments.GetArgument("hello");
Information("Hello arg value: {0}", helloArg);
```

```bash
dotnet cake --hello=Augusto
```

![image](https://user-images.githubusercontent.com/177608/107418777-71902e80-6aed-11eb-8e4b-d1c2d42281e8.png)
